chainstate/tx-verifier/src/transaction_verifier/transferred_amount_check.rs

```diff
@@ -84,13 +89,50 @@ where
     IssuanceTokenIdGetterFunc:
         Fn(&Id<Transaction>) -> Result<Option<TokenId>, ConnectTransactionError>,
 {
-    let inputs_total_map = calculate_total_inputs(
+    let mut inputs_total_map = calculate_total_inputs(
         utxo_view,
         pos_accounting_view,
         tx.inputs(),
         tx.get_id().into(),
         issuance_token_id_getter,
     )?;
+
+    let latest_token_version = chain_config
+        .chainstate_upgrades()
+        .version_at_height(current_height)
+        .1
+        .token_issuance_version();
+
+    match latest_token_version {
+        TokenIssuanceVersion::V0 => {
+            check_issuance_fee_burn_v0(chain_config, tx)?;
```

Comment:
I'm assuming that this will completely die after the fork.

Reply:
We will have to keep it in the testnet so that fee for older blocks is calculated correctly. But for the mainnet I hope this whole module will go away.